src/main/java/sda/training/rps/app/StageService.java

```diff
@@ -1,49 +1,141 @@
 package sda.training.rps.app;
 
-import sda.training.rps.dao.IStageDao;
 import sda.training.rps.dao.StageDao;
 import sda.training.rps.model.Game;
 import sda.training.rps.model.Stage;
 import sda.training.rps.util.Move;
 import sda.training.rps.util.Result;
 import sda.training.rps.util.ScannerSingleton;
 
-public class StageService implements IStageService {
-    private IStageDao stageDao = new StageDao();
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class StageService {
+    private final StageDao stageDao = new StageDao();
     private Move playerMove;
     private Move computerMove;
-    private Result[][] outcome;
-    private ScannerSingleton scanner = ScannerSingleton.getInstance();
+    public final Result[][] outcome = getOutcome();
+    private final ScannerSingleton scanner = ScannerSingleton.getInstance();
+    private final PlayerService playerService;
+
+    public StageService(PlayerService playerService) {
```

Comment:
jak nie dasz dao w konstruktorze to nie zrobisz testow :(
(gdyby to nie byl final moglbys teoretycznie wystawic setter do dao ale konstruktor + final jest o wiele lepszym pomyslem :)  )